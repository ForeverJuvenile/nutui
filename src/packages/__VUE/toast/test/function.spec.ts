import { ToastFunction } from './toast';

function sleep(delay = 0): Promise<void> {
  return new Promise((resolve) => {
    setTimeout(resolve, delay);
  });
}

describe('function toast', () => {
  test('show loading toast', async () => {
    ToastFunction.loading('加载中');
    expect(document.querySelector('.nut-toast.nut-toast-loading')).toBeTruthy();
    const text = document.querySelector('.nut-toast .nut-toast-text') as HTMLDivElement;
    expect(text.innerHTML).toEqual('加载中');
    await sleep(3000);
    const loading = document.querySelector('.nut-toast.nut-toast-loading');
    expect(loading).toBeNull();
  });

  test('show text toast', async () => {
    ToastFunction.text('文案', {
      duration: 50,
      title: '标题文字'
    });
    await sleep();
    const textToast = document.querySelector('.nut-toast.nut-toast-center') as HTMLElement;
    expect(textToast.innerHTML).toContain('标题文字');
    await sleep(100);
    expect(textToast.style.display).toEqual('none');
  });
  test('show fail toast', async () => {
    ToastFunction.fail('文案', {
      duration: 0
    });
    await sleep();
    const failToast = document.querySelector('.nut-toast');
    expect(failToast).toMatchSnapshot();
  });
  test('show warn toast', async () => {
    ToastFunction.warn('文案', {
      duration: 0
    });
    await sleep();
    const failToast = document.querySelector('.nut-toast');
    expect(failToast).toMatchSnapshot();
  });
});
