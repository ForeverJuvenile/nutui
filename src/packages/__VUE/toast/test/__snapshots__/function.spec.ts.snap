// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`function toast > show fail toast 1`] = `
<view
  class="nut-toast nut-toast-center nut-toast-has-icon nut-toast-base toast-fade-enter-from toast-fade-enter-active"
  iconsize="20"
  style="background-color: rgba(0, 0, 0, 0);"
>
  <view
    class="nut-toast-inner"
    style="text-align: center; background-color: rgba(0, 0, 0, 0.8);"
  >
    <view
      class="nut-toast-icon-wrapper"
    >
      <failure
        color="#ffffff"
      />
    </view>
    <!--v-if-->
    <view
      class="nut-toast-text"
    >
      文案
    </view>
  </view>
</view>
`;

exports[`function toast > show warn toast 1`] = `
<view
  class="nut-toast nut-toast-center nut-toast-has-icon nut-toast-base toast-fade-enter-from toast-fade-enter-active"
  iconsize="20"
  style="background-color: rgba(0, 0, 0, 0);"
>
  <view
    class="nut-toast-inner"
    style="text-align: center; background-color: rgba(0, 0, 0, 0.8);"
  >
    <view
      class="nut-toast-icon-wrapper"
    >
      <failure
        color="#ffffff"
      />
    </view>
    <!--v-if-->
    <view
      class="nut-toast-text"
    >
      文案
    </view>
  </view>
</view>
`;
